### PR TITLE
D8NID-969

### PR DESCRIPTION
**Pull request by johangant**

### CODE REVIEW

File: nidirect_news/src/Plugin/Block/FeaturedNewsBlock.php [hunk 1]
```diff
@@ -18,7 +20,9 @@ class FeaturedNewsBlock extends BlockBase {
    * {@inheritdoc}
    */
   public function build() {
-    $content['featured_news'] = \Drupal::service('nidirect_news.news')->getNewsEmbed('featured_news');
+    $view = \Drupal::service('nidirect_news.news')->getLatestNewsView();
```
Comment (DuttonMa) [inline nidirect_news/src/Plugin/Block/FeaturedNewsBlock.php:23]: Would it be better to use DI here ?

**Review by DuttonMa — APPROVED** [review]
Nice work, shame it had to be done & re-done !
Just a query about Dependency Injection

File: nidirect_news/src/Plugin/Block/FeaturedNewsBlock.php (hunk 1 above)
Comment (johangant) [reply nidirect_news/src/Plugin/Block/FeaturedNewsBlock.php:23]: Yep, in hindsight it couldn't hurt. Means a bit of extra boilerplate to go in but shouldn't take too long.

**Review by omahm — APPROVED** [review]
Looks good and as per Martin's comments, would be better to inject those services.

File: nidirect_news/src/Plugin/Block/FeaturedNewsBlock.php (hunk 1 above)
Comment (johangant) [reply nidirect_news/src/Plugin/Block/FeaturedNewsBlock.php]: Pushed an update this morning to sort this.

**Pull request merged**